wpilibc/shared/include/SmartDashboard/SendableChooser.inc

```diff
@@ -0,0 +1,127 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) FIRST 2011-2017. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace frc {
+
+template <class T>
+const std::string SendableChooser<T>::kDefault = "default";
+
+template <class T>
+const std::string SendableChooser<T>::kOptions = "options";
+
+template <class T>
+const std::string SendableChooser<T>::kSelected = "selected";
+
+/**
+ * Adds the given object to the list of options.
+ *
+ * On the {@link SmartDashboard} on the desktop, the object will appear as the
+ * given name.
+ *
+ * @param name   the name of the option
+ * @param object the option
+ */
+template <class T>
+void SendableChooser<T>::AddObject(const std::string& name, T& object) {
+  m_choices[name] = object;
+}
+
+/**
+ * Adds the given object to the list of options.
+ *
+ * On the {@link SmartDashboard} on the desktop, the object will appear as the
+ * given name.
+ *
+ * @param name   the name of the option
+ * @param object the option
+ */
+template <class T>
+void SendableChooser<T>::AddObject(const std::string& name, const T& object) {
+  m_choices[name] = const_cast<T*>(object);
```

Comment:
actually now that this is value rather than pointer based the const_cast shouldn't even be required